### DISCUSSION TITLE
New incidents or Service Requests are created when an existing case already exists.

### DISCUSSION THREAD

**Discussion by ServiceDeskSweden** [AdhocAdam/smletsexchangeconnector#174] [Q&A] [answered]
**Describe the bug**
When updating an existing SR or IR with an email to the workflow, the connector creates a new incident/SR but at the same time also updates email conversations in the existing IR. This seems to happen when the existing IR/SR has been changed in between the latest update. However the IR/SR is not open when the connector should update the case.

**Help us reproduce the bug**
Latest version of the connector
Features/variables you have enabled (i.e. $enableAzureCognitiveServicesForNewWI):

**Expected behavior**
The existing IR/SR should be updated, but a new IR/SR should not be created.

**Location and Environment**
Where are you running this from? Task Scheduler.
What SCSM version are you running? SCSM 2019. 10.19.1035.0

**Additional context**
Before upgrading to SCSM 2019 (SCSM v1807) this issue did not happen. After upgrading we changed to your connector scripts, and they are working very well, except this issue of "double IR/SR". You can see the attached file with the issue.
![SCSM](https://user-images.githubusercontent.com/64022062/79761537-e5283b80-8321-11ea-8ce8-4a8743e274c4.JPG)

Thanks for looking into this and your suggestions.

**Comment by AdhocAdam** [comment]
While I personally have not tested this against SCSM 2019 specifically and see no reason why it would be an issue, a few questions come to mind:

- What version of the connector?
- Do you have the **Merge Replies** feature enabled?
- What version of Exchange Web Services are you using?
- Any modifications done to the PowerShell (e.g. you introduced custom 3rd party ticket customizations) or was it download & deploy?
- Does this happen to every IR/SR or is only on a few IR/SRs? _If_ it's only a few my next question here is do those IR/SRs have any unique "shape" to them. For example, do they all have runbooks attached to them, is the email attached to them, etc.

It seems unlikely but this reminds me of a _very_ old issue that came about with the initial release of Merge Replies (I think that was maybe v1.1 in 2017?) wherein the initial sort of the inbox was done incorrectly so you had the double IR/SR thing.

**Comment by ServiceDeskSweden** [comment]
Thank you for your quick reply! See our answers below:

- What version of the connector?
**version 1.61**
- Do you have the Merge Replies feature enabled?
**No**
- What version of Exchange Web Services are you using?
**v14.3.32.0**
- Any modifications done to the PowerShell (e.g. you introduced custom 3rd party ticket customizations) or was it download & deploy?
**See the attached file (blurred = our company name)**
![1](https://user-images.githubusercontent.com/64022062/79861819-4e1cbb80-83d5-11ea-939f-283a5fb0b7fa.png)

-  Does this happen to every IR/SR or is only on a few IR/SRs? If it's only a few my next question here is do those IR/SRs have any unique "shape" to them. For example, do they all have runbooks attached to them, is the email attached to them, etc.
**This only happens on certain IR/SR's.** From what we can tell is that if we have added information in the newly created case (for example adding info in the action log or assigned the case to a technician), exit the IR/SR and then update the IR/SR with an email to the connector, it then creates the new IR/SR, but also updates the existing case. However we cannot be 100% certain that this is true, and see no real logic to when it is happening.
We have no runbooks attached, they are just plain IR/SR. 

BR

**Comment by AdhocAdam** [comment]
Actually for Exchange Web Services I was referring to the following path where x.x is the version number.

- C:\Program Files\Microsoft\Exchange\Web Services\x.x\

But I don't think that's relevant any more given your screenshot. It looks like you've introduced your own custom event within Update-WorkItem _after_ an Incident is updated. I'm assuming you've done the same for an SR as well? There shouldn't be an issue with doing that (because PowerShell!) but would it be possible for you to share the Type Projection and Template you're using here?

If you can't share - are these templates/type projections different than the ones from the ones that do the actual Work Item creation? If they are different _how_ different are they from one another? In that does the "update" one have something the "create" doesn't perhaps?

Lastly - if you have a development environment can you try to reproduce there and then manually trigger the connector so you can breakpoint/step through to see how/when this occuring?

**Comment by ServiceDeskSweden** [comment]
We'll get back to you on this ASAP!

**Comment by ServiceDeskSweden** [comment]
Would it be possible to upload or email our XML's somewhere directly to you Adam? The files do include a lot of information about our company's IR/SR and mailtemplates.

**Comment by AdhocAdam** [comment]
Yup, my secure email is over on my GitHub profile. Just click my picture over there to get it.

**Comment by ServiceDeskSweden** [comment]
Emailed!

**Comment by AdhocAdam** [comment, marked as answer]
It took some digging, but managed to identify and correct this behavior. The nature of the change can be viewed on #270. It will ship before the month is out with #275

**Comment by ServiceDeskSweden** [comment]
Hello Sir and hope you are well!

Many thanks for notifying us and that there is a new version of the connector coming with the fix.

Have a good day ahead Adam, and Take Care!

Best Regards, Richard / Perstorp ServiceDesk
